### ISSUE TITLE
"Value - Position" keyvalue for `comp_kv_setter`

### ISSUE THREAD

**Issue by Luke18033** [TeamSpen210/HammerAddons#234]
Add a keyvalue to `comp_kv_setter` that has the `origin` type, so the position is transformed when the entity is in an instance.

(This is needed for the final PEE update)

**Comment by TeamSpen210** [comment]
Seems useful, though I should try and figure out a better system, so there's not an option for every FGD value type...

**Comment by Luke18033** [comment]
Well it only needs to include separate KVs for things that are transformed by instances, so targetnames, positions, and angles, anything else can just be set as a string value

**Comment by TeamSpen210** [comment]
Hmm, this actually broke things, VBSP/Hammer will set this keyvalue when you move the entity around/collapse it, meaning the regular string is never used. Guess we need a mode choices field instead.

**Comment by vrad-exe** [comment]
Maybe you could make it where the other fields always take priority if set?

**Comment by TeamSpen210** [comment]
I could, but then you couldn't set it explicitly to an empty string if you needed that.

**Comment by vrad-exe** [comment]
Ah yeah

**Comment by piqey** [comment]
I'm making a `func_instance` for some meat hooks hanging by `phys_hinge`s in a map and I'm having trouble getting the axes to replicate properly.
![image](https://github.com/TeamSpen210/HammerAddons/assets/12129071/3ccf579a-b22a-484c-90d2-00566ef2bac0)
I'm pretty sure `comp_kv_setter` is the way I'll be able to fix this issue but I've yet to figure out how I'm supposed to configure it to fix `hingeaxis` on a per-instance basis.
![image](https://github.com/TeamSpen210/HammerAddons/assets/12129071/4d20fc06-0b97-453c-b3d4-4085e83fb346)
Any suggestions? Just throwing it out there; I've tried every combination I can think of.

**Comment by TeamSpen210** [comment]
If the `phys_hinge`'s keyvalue isn't being changed when the instance is collapsed, the `comp_kv_setter` isn't going to be transformed either.  Which game are you doing this in? It might be the case that Valve didn't implement instance functionality for the position keyvalue type. In that case, you'll need to move the KV setter outside of the instance and do it yourself manually.

**Comment by vrad-exe** [comment]
I think the current release build doesn't work right in older games but I fixed it in dev

**Comment by vrad-exe** [comment]
The type needs to be `origin` for it to get transformed I'm pretty sure, but the current release has it as like `vector` or something

**Comment by piqey** [comment]
> If the `phys_hinge`'s keyvalue isn't being changed when the instance is collapsed, the `comp_kv_setter` isn't going to be transformed either. Which game are you doing this in? It might be the case that Valve didn't implement instance functionality for the position keyvalue type. In that case, you'll need to move the KV setter outside of the instance and do it yourself manually.

It's Source SDK 2013 Multiplayer.

> I think the current release build doesn't work right in older games but I fixed it in dev

> The type needs to be `origin` for it to get transformed I'm pretty sure, but the current release has it as like `vector` or something

I ought to switch builds then, my thanks

**Comment by piqey** [comment]
Having issues compiling the dev branch, though—can't build `postcompiler.exe`. Stuck on this error:
```
Error: Invalid hiddenimport 'srctools._pyinstaller.hook-srctools'. Hidden imports should be importable module names – not file paths. i.e. use --hiddenimport=foo.bar instead of --hiddenimport=.../site-packages/foo/bar.py
```
Everything else seems to work fine. Tried on both Python 3.12 and 3.11. Should I be using an earlier version?

**Comment by TeamSpen210** [comment]
Yeah that's a bug with the latest version of PyInstaller, try version `6.1.0`? But the change here is just in the FGD for `comp_kv_setter`, you don't need the different compiler build.

**Comment by piqey** [comment]
> Yeah that's a bug with the latest version of PyInstaller, try version `6.1.0`? But the change here is just in the FGD for `comp_kv_setter`, you don't need the different compiler build.

Yep, that was it. Got it built now. Will give `comp_kv_setter` a shot soon.
